### PR TITLE
Mds 3714

### PR DESCRIPTION
**Pull request by wilson496**
# Main

- Able to update the mine list of a minespace user in Core-web
 - this includes being able to add new mines and delete existing mines

# Other

- When updating a proponent, the pop-up will read Update User and Edit Proponent to correctly represent the form as updating the user. 

# How to test

- Create 2-3 mines
- Create multiple proponents in Core (Navigate to Admin -> MineSpace Management -> Users -> Create Proponent)
- Add and delete mines across the proponents you have created
- Data should update in real-time without the need for a hard refresh

# Notes

- N/A

### CODE REVIEW

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (BenHarrington) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py]: To ensure the client handles an error correctly raise an InternalServerError with that error message rather than a console print. That error is found in the werkzeug.exceptions namespace.

File: services/core-web/src/components/admin/MinespaceUserList.js
Comment (BenHarrington) [inline services/core-web/src/components/admin/MinespaceUserList.js:59]: Name and alt attributes should reflect edit, not remove.

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (BenHarrington) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py]: Remove this before merge.

File: services/core-api/app/api/users/minespace/models/minespace_user.py
Comment (BenHarrington) [inline services/core-api/app/api/users/minespace/models/minespace_user.py]: We should be raising a NotFoundError if minespace_user == None before attempting to save.

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (BenHarrington) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py]: The PUT method should be contained within the MinespaceUserResource rather than the MinespaceUserListResource class given how we have the namespaces configured. The list resources typically handle GET ALL and POST, whereas the base resource handles operations on a single entity such as GET, PUT, DELETE.

Additionally you will need to add in the request parser into the MinespaceUserResource, much the same way that it is being done in the MinespaceUserListResource. 

Let me know if you need a walkthrough or have any questions.

File: services/core-web/src/components/modalContent/UpdateMinespaceUserModal.js
Comment (BenHarrington) [inline services/core-web/src/components/modalContent/UpdateMinespaceUserModal.js]: When clicking the edit/pencil icon on an existing minespace users record the modal secondary header reads "Create Proponent", this should be "Edit Proponent". In addition to that, the Email/BCeID username and Mines fields do not pre-populate with the existing record.

File: services/core-web/src/components/admin/MinespaceUserManagement.js
Comment (BenHarrington) [inline services/core-web/src/components/admin/MinespaceUserManagement.js:137]: When clicking on the Delete icon for an existing minespace users record it displays the confirm text "Are you sure you want to delete undefined?", this is a known issue in Prod but I believe the reference just needs to be updated, it currently is "record.email" when I believe it should be "record.email_or_username". This isn't required for this ticket but would be a nice quick win if you are able.

The code diff mentioned above is not where the problem exists, there were no changes in the PR so I couldn't attach it in the PR.

**Review by BenHarrington — CHANGES_REQUESTED** [review]
Definitely on the right track here Cameron. Reach out to me if you have questions.

File: services/core-api/app/api/users/minespace/models/minespace_user_mine.py
Comment (mmumby) [inline services/core-api/app/api/users/minespace/models/minespace_user_mine.py:31]: delete the logger and print statement here

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (mmumby) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py:76]: remove logger

File: services/core-web/src/tests/components/admin/MinespaceUserManagement.spec.js
Comment (mmumby) [inline services/core-web/src/tests/components/admin/MinespaceUserManagement.spec.js:23]: This was added twice - delete one.

File: services/core-web/src/tests/components/modalContent/UpdateMinespaceUserModal.spec.js
Comment (mmumby) [inline services/core-web/src/tests/components/modalContent/UpdateMinespaceUserModal.spec.js:4]: This should be using the function/class import instead of the default. This component isn't connected to redux, so the test is passing, but if it was a connected component the test would fail. It's a good habit to only use the class/functional import when testing components. 

```import { UpdateMinespaceUserModal } from "@/components/modalContent/UpdateMinespaceUserModal"```

File: services/core-web/src/components/admin/MinespaceUserManagement.js
Comment (mmumby) [inline services/core-web/src/components/admin/MinespaceUserManagement.js:75]: Can you update the title to specify what user is being edited? 

`Update User: ${record.email_or_username}`

File: services/core-web/src/components/admin/UpdateMinespaceUser.js
Comment (mmumby) [inline services/core-web/src/components/admin/UpdateMinespaceUser.js:22]: This doesn't appear to be getting used in this file: 'refreshData', did you have linting errors? It can be removed if not used, and removed from the 'OpenModal' function

File: services/core-web/src/components/admin/MinespaceUserManagement.js
Comment (mmumby) [inline services/core-web/src/components/admin/MinespaceUserManagement.js:108]: The component doesn't appear to be using this: refreshData
remove if not needed.

File: services/core-web/src/components/admin/MinespaceUserList.js
Comment (mmumby) [inline services/core-web/src/components/admin/MinespaceUserList.js:63]: Thanks for making that fix!

File: services/core-web/src/components/Forms/EditMinespaceUser.js
Comment (mmumby) [inline services/core-web/src/components/Forms/EditMinespaceUser.js]: This validation isn't needed in the Edit Modal as the username cannot be edited. Please remove.

File: services/core-web/src/components/Forms/EditMinespaceUser.js
Comment (mmumby) [inline services/core-web/src/components/Forms/EditMinespaceUser.js:17]: Once the validation is removed, this prop is no longer needed. 'minespaceUserEmailHash', please remove, and remove from being passed into the 'OpenModal' function

**Review by mmumby — DISMISSED** [review]
Nice work! I have a few comments on the FE to clean up the code, and I'm wondering if your linter is working? I've noticed a few things that should have been caught. If you used '--no-verify' at any point, please run the linter manually and fix anything related to the files you changed. 

Justin will review the BE, and I'm happy with the manual testing we did together. I'll give it another quick test when its code complete.

File: services/core-web/src/tests/components/admin/MinespaceUserManagement.spec.js
Comment (mmumby) [reply services/core-web/src/tests/components/admin/MinespaceUserManagement.spec.js:23]: 'updateMinespaceUserMines'

File: services/core-api/app/api/users/minespace/models/minespace_user_mine.py
Comment (JustinMacaulay) [inline services/core-api/app/api/users/minespace/models/minespace_user_mine.py]: unused import?

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (JustinMacaulay) [reply services/core-api/app/api/users/minespace/resources/minespace_user.py:76]: if you're using current_app and logging to info, warning, and so on its probably fine to leave them in its just debug that I would remove.

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (JustinMacaulay) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py]: I would change this to info or a higher level of logging.

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (JustinMacaulay) [inline services/core-api/app/api/users/minespace/resources/minespace_user.py]: this put is updating a specific user and its associated mines so I am not sure why the return statement is all minespaceusers.

**Review by JustinMacaulay — COMMENTED** [review]
just a few comments on the backend stuff so far

**Review by mmumby — COMMENTED** [review]
commented on the wrong Pr - cant delete. disregard.

File: services/core-api/app/api/users/minespace/resources/minespace_user.py
Comment (wilson496) [reply services/core-api/app/api/users/minespace/resources/minespace_user.py]: Fixed.

**Review by mmumby — COMMENTED** [review]
Looks great, works great, thanks for making those changes! Happy to approve if Justins comments are resolved.

**Review by BenHarrington — APPROVED** [review]
Great work Cam!

**Pull request merged**
